ui/src/telegrafs/components/CollectorCard.tsx

```diff
@@ -63,52 +53,46 @@ class CollectorRow extends PureComponent<Props & WithRouterProps> {
     const {collector, bucket} = this.props
 
     return (
-      <>
-        <IndexList.Row>
-          <IndexList.Cell>
-            <ComponentSpacer
-              margin={ComponentSize.Small}
-              direction={FlexDirection.Column}
-              alignItems={AlignItems.FlexStart}
-              stretchToFitWidth={true}
-            >
-              <EditableName
-                onUpdate={this.handleUpdateName}
-                name={collector.name}
-                noNameString={DEFAULT_COLLECTOR_NAME}
-                onEditName={this.handleNameClick}
-              />
-              <EditableDescription
-                description={collector.description}
-                placeholder={`Describe ${collector.name}`}
-                onUpdate={this.handleUpdateDescription}
-              />
-              {this.labels}
-            </ComponentSpacer>
-          </IndexList.Cell>
-          <IndexList.Cell>{bucket}</IndexList.Cell>
-          <IndexList.Cell revealOnHover={true} alignment={Alignment.Right}>
-            <ComponentSpacer
-              margin={ComponentSize.Small}
-              direction={FlexDirection.Row}
-              justifyContent={JustifyContent.FlexEnd}
-            >
-              <Button
-                text="Setup Instructions"
-                size={ComponentSize.ExtraSmall}
-                color={ComponentColor.Secondary}
-                onClick={this.handleOpenInstructions}
-              />
-              <ConfirmationButton
-                size={ComponentSize.ExtraSmall}
-                text="Delete"
-                confirmText="Confirm"
-                onConfirm={this.handleDeleteConfig}
-              />
-            </ComponentSpacer>
-          </IndexList.Cell>
-        </IndexList.Row>
-      </>
+      <ResourceList.Card
+        key={`telegraf-id--${collector.id}`}
+        testID="resource-card"
+        name={() => (
+          <ResourceList.Name
+            onUpdate={this.handleUpdateName}
+            onClick={this.handleNameClick}
+            name={collector.name}
+            noNameString={DEFAULT_COLLECTOR_NAME}
+            parentTestID="collector-card--name"
+            buttonTestID="collector-card--name-button"
+            inputTestID="collector-card--input"
+          />
+        )}
+        description={() => (
+          <ResourceList.Description
+            onUpdate={this.handleUpdateDescription}
+            description={collector.description}
+            placeholder={`Describe ${collector.name}`}
+          />
+        )}
+        labels={() => this.labels}
+        metaData={() => [
+          <>Bucket: {bucket}</>,
+          <>
+            <a onClick={this.handleOpenInstructions}>Setup Instructions</a>
```

Comment:
you'll have to add `href="#"` for the link to receive the proper styles. The other way you could do this is use a `<Link />` element and route to the path for the overlay